clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1423,34 +1422,40 @@ public void resume(Collection<TopicPartition> partitions) {
                 throw new IllegalArgumentException("The target time for partition " + entry.getKey() + " is " +
```

Comment:
Internally, `Fetcher` may have timestamp = -1 in the `OffsetAndTimestamp` when getting the latest offsets. So adding check in the constructor may not work. But the -1 timestamp won't be exposed to the users. If the concern is that we are going to return a -1 timestamp to the users, we can do a final round of check before returning the result to the users. I do not have a strong opinion on this, it is a defensive coding to detect a potential bug in our code if that happens, although personally I think that should be covered by unit test instead of in the code.